upgrade_test.py

```diff
@@ -1058,10 +1058,6 @@ def test_kubernetes_platform_upgrade(self):
         upgrade_version, scylla_pool = self.k8s_cluster.upgrade_kubernetes_platform(
             pod_objects=self.db_cluster.nodes,
             use_additional_scylla_nodepool=True)
-        # NOTE: refresh IP addresses of the Scylla nodes because they get changed in current case
-        for db_node in self.db_cluster.nodes:
-            db_node.refresh_ip_address()
-        self.monitors.reconfigure_scylla_monitoring()
```

Comment:
lets remove all the hacks, also from nemesis, as we talked

Reply:
Did it in my latest dev update which has not been uploaded here yet. 
But remember, if this thread gets stuck, we won't have any callbacks be running.
So, it is kind of risky to remove it from my POW.

Reply:
I still think we should remove those, and not do those extra steps again and again.

Reply:
Done